### PR TITLE
[Messenger] Added an implementation to the TransportInterface

### LINKED ISSUE #12230: [Messenger] Document "own Receiver"
it's possible someone explain "own receiver" with a case.
The doc just show  the class who implement ReceiverInterface and not how to use it...
thank a lot

### PR DESCRIPTION
**Pull request by Nyholm**
This will fix #12230

This is basically a much simpler `DoctrineTransport` with easy query builders and no real error handling.

### CODE REVIEW

File: messenger/custom-transport.rst [hunk 1]
```diff
@@ -31,33 +31,111 @@ DSN. You will need a transport factory::
...
+            return $envelope->with(new TransportMessageIdStamp($uuid));
         }
     }
 
+.. note::
+
+    The implementation above is not runnable code but illustrates how a
+    :class:`Symfony\\Component\\Messenger\\Transport\\TransportInterface` could
+    be implemented. For real implementations see :class:`Symfony\\Component\\Messenger\\Transport\\InMemoryTransport`
+    and :class:`Symfony\\Component\\Messenger\\Transport\\Doctrine\\DoctrineReceiver`.
```
Comment (weaverryan) [inline messenger/custom-transport.rst]: I think we can remove the note. What I mean is - just make this a real paragraph here.

**Review by weaverryan — COMMENTED** [review]
I like the balance between a simplified, real-ish example and linking to the core code

File: messenger/custom-transport.rst (hunk 1 above)
Comment (Nyholm) [reply messenger/custom-transport.rst]: Thank you.

**Comment by Nyholm** [comment]
Thank you for the feedback. PR is updated

**Comment by javiereguiluz** [comment]
@Nyholm thanks a lot for this contribution! Should this be merged in 4.3 or 4.4? Thanks.

**Comment by Nyholm** [comment]
I just checked. We can merge this into 4.3.

**Comment by javiereguiluz** [comment]
Nice addition Tobias! It's merged (in 4.3) now.

**Comment by Nyholm** [comment]
Thank you for merging.

**Comment by xfifix** [comment]
@Nyholm  Thanks a lot for

**Pull request merged**